File: console/src/main/kotlin/app/accrescent/parcelo/console/data/net/Draft.kt
// Copyright 2023 Logan Magee
//
// SPDX-License-Identifier: AGPL-3.0-only

package app.accrescent.parcelo.console.data.net

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Draft(
    val id: String,
    @SerialName("app_id")
    val appId: String,
    val label: String,
    @SerialName("version_code")
    val versionCode: Int,
    @SerialName("version_name")
    val versionName: String,
    @SerialName("creation_time")
    val creationTime: Long,
    val status: DraftStatus,
)

@Serializable
enum class DraftStatus {
    @SerialName("unsubmitted")
    UNSUBMITTED,

    @SerialName("submitted")
    SUBMITTED,

    @SerialName("approved")
    APPROVED,

    @SerialName("rejected")
    REJECTED,

    @SerialName("publishing")
    PUBLISHING,
}

File: console/frontend/src/app/draft.ts
// Copyright 2023 Logan Magee
//
// SPDX-License-Identifier: AGPL-3.0-only

export interface Draft {
    id: string;
    app_id: string;
    label: string;
    version_code: number;
    version_name: string;
    creation_time: number;
    status: DraftStatus;
}

export enum DraftStatus {
    Unsubmitted = 'unsubmitted',
    Submitted = 'submitted',
    Approved = 'approved',
    Publishing = 'publishing',
}
